src/Microsoft.TemplateEngine.Utils/CryptoRandom.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Microsoft.TemplateEngine.Utils
+{
+    public class CryptoRandom : Random
+    {
+        private RNGCryptoServiceProvider _rng = new RNGCryptoServiceProvider();
+        private byte[] _uint32Buffer = new byte[4];
+
+        public CryptoRandom() { }
+
+        public CryptoRandom(int ignoredSeed) { }
+
+        public override int Next()
+        {
+            _rng.GetBytes(_uint32Buffer);
+            return BitConverter.ToInt32(_uint32Buffer, 0) & 0x7FFFFFFF;
+        }
+
+        public override int Next(int maxValue)
+        {
+            if (maxValue < 0) { throw new ArgumentOutOfRangeException(nameof(maxValue)); }
+            return Next(0, maxValue);
+        }
+
+        public override int Next(int minValue, int maxValue)
+        {
+            if (minValue > maxValue) { throw new ArgumentOutOfRangeException(nameof(minValue)); }
+            if (minValue == maxValue) { return minValue; }
+            long diff = maxValue - minValue;
+            // We are avoiding reminder bias by discarding the reminder.
+            //  background: https://ericlippert.com/2013/12/16/how-much-bias-is-introduced-by-the-remainder-technique/
+            while (true)
+            {
+                _rng.GetBytes(_uint32Buffer);
+                uint rand = BitConverter.ToUInt32(_uint32Buffer, 0);
+                long max = (1 + (long)uint.MaxValue);
+                long remainder = max % diff;
+                if (rand < max - remainder)
+                {
+                    return (int)(minValue + (rand % diff));
+                }
+            }
+        }
+
+        public override double NextDouble()
+        {
+            _rng.GetBytes(_uint32Buffer);
+            uint rand = BitConverter.ToUInt32(_uint32Buffer, 0);
+            return rand / (1.0 + uint.MaxValue);
+        }
+
+        public override void NextBytes(byte[] buffer)
+        {
+            if (buffer == null) { throw new ArgumentNullException(nameof(buffer)); }
```

Comment:
Nit: Seems like we are introducing a new code style to the repo. As long as you agree as the contributors, I think that is fine. Just wanted to point out :)

"Back in my day" (how did I get old so quick? :-0 ), the "repo consistent way" would probably be one of these:

``` 
buffer = buffer ?? throw new ArgumentNullException();
```

or
```
if (buffer == null)
{
    throw new ....
}```

Reply:
I clicked on the drop-down for .net framework 4.7.2, and completely missed the purple banner at the top that said I'd been automatically redirected. I _really_ hate that experience on the docs site.